### ISSUE TITLE
Duplicates

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#4120]
This is for a bug fix.  I assigned to Maxwell, so he can re-assign.

Duplicates works well generally.  But it doesn't work (for me) when a filter is applied.  I think any filter.

I tried with a new worksheet.  I added a column V3 to the data frame (called data1) with the values 1, 2, ... 10.  Then filtered out the values <=3.  Then used the duplicates dialogue:

It says:

Error running R command(s)

Error in self$get_data_objects(data_name)$add_columns_to_data(col_name,  : 
  Length of new column must be divisible by the length of the data frame
The error occurred in attempting to run the following R command(s):

dup1 <- questionr::duplicated2(x=InstatDataObject$get_columns_from_data(data_name="data1", col_names="Regular1"))

InstatDataObject$add_columns_to_data(col_data=dup1, col_name="dup1", data_name="data1")

OK

**Comment by dannyparsons** [comment]
This is easy to prevent the error. We need to add the same option as we do with all other dialogs which add a column to the data. I believe it's an option set on the selector or receiver.
This would mean the filter is ignored when creating the new column so it's of the right length. That might not be what you want to happen here though.

**Comment by rdstern** [comment]
I'd be happy for it here, i.e. for filter to be ignored for this command.
There are other commands - particularly our calculator - that also ignore filters.  This is more complicated and needs more discussion.  I mention that in a further issue.

**Comment by Patowhiz** [comment]
@maxwellfundi  Let me have a look at it

**Comment by dannyparsons** [comment]
Yes ignoring the filter isn't useful here either. We need further options in our R code to be able to calculate on the filtered data and add back only calculated values.

**Comment by Patowhiz** [comment]
@dannyparsons please revert back to this issue if a solution in R code is found. Having read at your comments it seems like this is an R issue which I'm still grasping. Will be following up on how this is solved in R.

**Comment by Patowhiz** [comment]
@dannyparsons please revert back to this issue if a solution in R code is found. I haven't tried solving it but according to rdstern and your comments it seems like this is an R issue which I'm still learning about. Will be following up on how this is solved in R.

**Comment by dannyparsons** [comment]
Yes let's leave this open and solve within the R code.

**Comment by lilyclements** [comment]
Since this is an issue with the R code, I've moved this to it's new home in the databook package [here](https://github.com/IDEMSInternational/databook/issues/76)
